src/app/courses/course-item/course-item.component.spec.ts

```diff
@@ -1,6 +1,60 @@
 import { async, ComponentFixture, TestBed } from '@angular/core/testing';
 
-import { CourseItemComponent } from './course-item.component';
+import { CourseItemComponent, ICourse } from './course-item.component';
+import { SimpleChange, Component } from '@angular/core';
+import { By } from '@angular/platform-browser';
+
+@Component({
+  template: `<angular-course-item (onDeleteCourse)="onDeleteCourse($event)" [item]="item">
+  </angular-course-item>`
+})
+class TestHostComponent {
+  public item = {
+    id: 1,
+    title: "title",
+    creationDate: new Date(),
+    durationMin: 0,
+    description: "description"
+  };
+  public itemToDelete: number;
+  public onDeleteCourse(currentItem: number) {
+    this.itemToDelete = currentItem;
+  } 
+}
+
+describe('TestHostComponent', () => {
+  let testHost: TestHostComponent;
+  let fixture: ComponentFixture<TestHostComponent>;
+  let item: ICourse;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [ CourseItemComponent, TestHostComponent ]
+    })
+    .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(TestHostComponent);
+    testHost = fixture.componentInstance;
+    fixture.detectChanges();
+  });  
+
+  it('should emit delete item', () => {
+    fixture.detectChanges();
```

Comment:
we detect changes before each test so this detection looks useless

Reply:
Removed this call and other useless ones